sound/soc/sof/intel/mtl.c

```diff
@@ -605,6 +605,36 @@ static u64 mtl_dsp_get_stream_hda_link_position(struct snd_sof_dev *sdev,
 	return ((u64)llp_u << 32) | llp_l;
 }
 
+static int mtl_dsp_core_get(struct snd_sof_dev *sdev, int core)
+{
+	const struct sof_ipc_pm_ops *pm_ops = sdev->ipc->ops->pm;
+
+	if (core == SOF_DSP_PRIMARY_CORE)
+		return mtl_dsp_core_power_up(sdev, SOF_DSP_PRIMARY_CORE);
+
+	if (pm_ops->set_core_state)
+		return pm_ops->set_core_state(sdev, core, true);
```

Comment:
so if this fails you will leave the primary core powered-up, is this intentional?

Reply:
primary core is powered off when we do pm suspend and if primary core is powered off host can't access audio dsp.   If there is a stream working on primary core, and another stream try to work on secondary core but get some error, we can't power off primary core.